src/components/pages/devices/devices.js

```diff
@@ -38,19 +42,26 @@ export class Devices extends Component {
     selectedDeviceId: id
   });
 
+  onContextMenuChange = contextBtns => this.setState({ contextBtns });
+
   getSoftSelectId = ({ id }) => id;
 
   render() {
     const { t, devices, error, isPending, lastUpdated, entities, fetchDevices } = this.props;
     const gridProps = {
       rowData: isPending ? undefined : devices || [],
       onSoftSelectChange: this.onSoftSelectChange,
+      onContextMenuChange: this.onContextMenuChange,
       softSelectId: this.state.selectedDeviceId,
       getSoftSelectId: this.getSoftSelectId,
       t: this.props.t
     };
-    return (
-      <div className="devices-container">
+    return [
+      <ContextMenu>
+        { this.state.contextBtns }
+        <Btn svg={svgs.plus}>New device</Btn>
```

Comment:
Needs to be translated, this is only here as an example

Reply:
Add a TODO